### PR TITLE
Yet more more spkgs confs

### PR DESCRIPTION
**Pull request by dimpase**
a continuation of #36276  (more system Python packages support via `--enable-system-site-packages`)

Also, added a Cython 3.0.2 patch to fix the usage of `--enable-system-site-packages` option. Without it, and sufficiently many Python system packages, sagelib won't build. (Therefore, blocker label added)

### CODE REVIEW

**Comment by dimpase** [comment]
need to add more distros/ stuff

**Comment by orlitzky** [comment]
We could do cython now too. The standard `spkg-configure.m4` works fine. I've only tested it with cython-3.x but we'll probably have that soon from https://github.com/sagemath/sage/pull/36110

**Comment by mkoeppe** [comment]
Can you please redo this branch with proper merge commits for its depencies?

**Comment by dimpase** [comment]
by right, this branch should be based off develop, but it needs #36276, which is only in  the "WIP-develop".
This gives me a headache. Anyhow, it's all in
https://github.com/dimpase/sage/tree/pr36332

**Comment by mkoeppe** [comment]
I'd suggest to remove the additions to hatch_nodejs_version because #36129 will remove this package.

**Comment by mkoeppe** [comment]
Or let's not lay eggs that are unwanted?

**Comment by dimpase** [comment]
one cannot realistically scan all the open PRs and check for potential conflicts

**Comment by mkoeppe** [comment]
I know, hence my helpful comment

**Comment by mkoeppe** [comment]
@orlitzky In the platforms tests for `--enable-system-site-packages`, I see Cython compiler crashes while building sagelib, on many platforms. 
https://github.com/mkoeppe/sage/actions/runs/6360039461/job/17277499571#step:12:58476
```
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      # distutils: libraries = NTL_LIBRARIES gmp
      ^
      ------------------------------------------------------------
  
      sage/rings/polynomial/polynomial_integer_dense_flint.pyx:1:0: Compiler crash in DebugTransform
  
  
      Compiler crash traceback from this point on:
        File "Cython/Compiler/Visitor.py", line 182, in Cython.Compiler.Visitor.TreeVisitor._visit
        File "/sage/local/var/lib/sage/venv-python3.10/lib/python3.10/site-packages/Cython/Compiler/ParseTreeTransforms.py", line 4053, in visit_ModuleNode
          self.visit_FuncDefNode(nested_funcdef)
        File "/sage/local/var/lib/sage/venv-python3.10/lib/python3.10/site-packages/Cython/Compiler/ParseTreeTransforms.py", line 4113, in visit_FuncDefNode
          self.tb.start(arg.name)
        File "/sage/local/var/lib/sage/venv-python3.10/lib/python3.10/site-packages/Cython/Debugger/DebugWriter.py", line 42, in start
          self.tb.start(name, attrs or {})
        File "src/lxml/saxparser.pxi", line 841, in lxml.etree.TreeBuilder.start
        File "src/lxml/saxparser.pxi", line 769, in lxml.etree.TreeBuilder._handleSaxStart
        File "src/lxml/apihelpers.pxi", line 179, in lxml.etree._makeSubElement
        File "src/lxml/apihelpers.pxi", line 1753, in lxml.etree._tagValidOrRaise
      ValueError: Invalid tag name '.0'
      Traceback (most recent call last):
        File "/sage/local/var/lib/sage/venv-python3.10/lib/python3.10/site-packages/Cython/Build/Dependencies.py", line 1325, in cythonize_one_helper
          return cythonize_one(*m)
        File "/sage/local/var/lib/sage/venv-python3.10/lib/python3.10/site-packages/Cython/Build/Dependencies.py", line 1301, in cythonize_one
          raise CompileError(None, pyx_file)
      Cython.Compiler.Errors.CompileError: sage/rings/polynomial/polynomial_integer_dense_flint.pyx
```

Any  thoughts?

**Comment by orlitzky** [comment]
> Any thoughts?

Looks like https://github.com/cython/cython/issues/5552

**Comment by orlitzky** [comment]
Unrelated: `dev-python/stack_data` was just renamed to `dev-python/stack-data` in Gentoo so now its `distros/gentoo.txt` needs an update.

File: build/pkgs/calver/distros/package-version.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+2022.6.26
```
Comment (orlitzky) [inline build/pkgs/calver/distros/package-version.txt:1]: What's this do?

**Comment by dimpase** [comment]
calver is for setting up python package versions via date (CalendarVersion)
And, well, the version Sage installs is 2022.6.26
(same version as in Gentoo a.t.m.)

**Comment by orlitzky** [comment]
> calver is for setting up python package versions via date (CalendarVersion) And, well, the version Sage installs is 2022.6.26 (same version as in Gentoo a.t.m.)

But why create a `package-version.txt` under `distros/`?

**Comment by dimpase** [comment]
this happends when you put a bunch of .txt files in a wrong place, then move them to the right place, but...

**Comment by dimpase** [comment]
> > Any thoughts?
> 
> Looks like [cython/cython#5552](https://github.com/cython/cython/issues/5552)

I tried patching Cython with the patch they provide, still getting this error.

**Comment by orlitzky** [comment]
> PS. Sorry, it's OK - I forgot how to patch Gentoo packages, and the patch got lost...

I opened a Gentoo bug to backport the patch. Otherwise cython-3.0.3 should fix it, or `--without-system-cython`. I'm happy with this branch if Matthias has no more comments.

**Comment by dimpase** [comment]
we'd also sort out these endless
```
Implicit noexcept declaration is deprecated. Function declaration should contain 'noexcept' keyword.
```
warnings, eventually

**Comment by orlitzky** [comment]
I don't mind the cython patch but note that it won't help when cython also comes from the system (after this PR). The cython bug you patched is fixed in cython-3.0.3 (https://github.com/sagemath/sage/pull/36416), but cython-3.0.3 has a new problem (https://github.com/cython/cython/issues/5748). Hopefully 3.0.4 comes soon.

**Comment by mkoeppe** [comment]
Note that #36411 pins Cython to 3.0.2; waiting for review.

**Comment by dimpase** [comment]
Can this PR be approved? Any remaining problems with it?

**Comment by dimpase** [comment]
> I don't mind the cython patch but note that it won't help when cython also comes from the system (after this PR). 
the system better be carrying this patch, what can I say...

**Pull request merged**